client/test/unit/components/bhLocationSelect.spec.js

```diff
@@ -0,0 +1,201 @@
+
+    {uuid : '0x03A329B203FE4F73B40F56A2870CC7F8', name : 'village 17', sector_uuid : '0x2C90F3C84EB146D39A5CDA180D1430C4'},
+    {uuid : '0x03A329B203FE4F73B40F56A2870CC7F9', name : 'village 18', sector_uuid : '0x2C90F3C84EB146D39A5CDA180D1430C4'},
+
+    {uuid : '0x03A329B203FE4F73B40F56A2870CC7A1', name : 'village 19', sector_uuid : '0x2C90F3C84EB146D39A5CDA180D1430C5'},
+    {uuid : '0x03A329B203FE4F73B40F56A2870CC7A2', name : 'village 20', sector_uuid : '0x2C90F3C84EB146D39A5CDA180D1430C5'},
+
+    {uuid : '0x03A329B203FE4F73B40F56A2870CC7A3', name : 'village 21', sector_uuid : '0x2C90F3C84EB146D39A5CDA180D1430C6'},
+    {uuid : '0x03A329B203FE4F73B40F56A2870CC7A4', name : 'village 22', sector_uuid : '0x2C90F3C84EB146D39A5CDA180D1430C6'},
+
+    {uuid : '0x03A329B203FE4F73B40F56A2870CC7A5', name : 'village 23', sector_uuid : '0x2C90F3C84EB146D39A5CDA180D1430C7'},
+    {uuid : '0x03A329B203FE4F73B40F56A2870CC7A6', name : 'village 24', sector_uuid : '0x2C90F3C84EB146D39A5CDA180D1430C7'},
+  ]
+};
+
+describe('component Template and controller : bhLocationSelect', function () {
+  var componentController, element, scope, $compile;
+  var template = `<bh-location-select id="location-id" location-uuid="testData.villages[0].uuid"></bh-location-select>`;
+
+  beforeEach(module('pascalprecht.translate', 'ngStorage', 'angularMoment', 'ui.bootstrap', 'bhima.services', 'bhima.components', 'templates'));
+
+  beforeEach(inject(function (_$rootScope_, _$compile_, _$httpBackend_) {
+    scope = _$rootScope_.$new();
+    $compile = _$compile_;
+    $httpBackend = _$httpBackend_;
+
+    $httpBackend.whenGET('/locations/countries').respond(testData.countries);
+    $httpBackend.whenRoute('GET', '/locations/detail/:uuid')
+      .respond(function (method, url, data, headers, params) {
+        var response = {};
+
+        var village =  testData.villages.filter(function (village){ return village.uuid === params.uuid; })[0];
+        var sector = testData.sectors.filter(function (sector){ return sector.uuid === village.sector_uuid; })[0];
+        var province = testData.provinces.filter(function (province){ return province.uuid === sector.province_uuid; })[0];
+        var country = testData.countries.filter(function (country){ return country.uuid === province.country_uuid; })[0];
+
+        response.villageUuid = village.uuid;
+        response.village = village.name;
+        response.sector = sector.name;
+        response.sectorUuid = sector.uuid;
+        response.province = province.name;
+        response.provinceUuid = province.uuid;
+        response.country = country.name;
+        response.countryUuid = country.uuid;
+
+        return [200, response];
+      });
+    $httpBackend.whenRoute('GET', '/locations/provinces')
+      .respond(function (method, url, data, headers, params) {
+        var response;
+        if(params.country){
```

Comment:
Instead of having an `if ... else` statement in the `$httpBackend` response, it would be a much stronger test to write two different `$httpBackend` responses.  One response for when the `country_uuid` has been specified, and another for when the `country_uuid` has not been specified.